presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -45,7 +46,8 @@
     private long currentPosition;
     private long currentGroupRowCount;
     private long nextRowInGroup;
-    private Map<ColumnDescriptor, ParquetColumnReader> columnReadersMap = new HashMap<>();
+    private long currentRowGroup;
+    private Map<ColumnIndex, ParquetColumnReader> columnReadersMap = new HashMap<>();
```

Comment:
For a big file will this map become huge?  If so, maybe clear the map in advanceToNextRowGroup